apps/ensindexer/src/config/types.ts

```diff
@@ -0,0 +1,236 @@
+   *
+   * Invariants:
+   * - A set of valid plugin names with at least one value
+   * - For any requested plugin, the config will have a key for chainId in the
+   *   indexedChains object. In that indexedChains object, the config will have a
+   *   rpcEndpointUrl that is defined for that chainId and a rpcMaxRequestsPerSecond
+   *   that is defined for that chainId. This is ensured by the validateChainConfigs
+   *   function in `validations.ts` and not part of the schema validation here.
+   */
+  plugins: Set<PluginName>;
+
+  /**
+   * A feature flag to enable or disable healing of addr.reverse subnames.
+   * If this is set to true, ENSIndexer will attempt to heal subnames of
+   * addr.reverse.
+   *
+   * If this is not set, the default value is set to `DEFAULT_HEAL_REVERSE_ADDRESSES`.
+   *
+   * Setting this to `true` results in indexed data no longer being backwards
+   * compatible with the ENS Subgraph. For full data-level backwards
+   * compatibility with the ENS Subgraph, this should be set to `false`.
+   *
+   * Invariant: The value must be 'true' or 'false'
+   */
+  healReverseAddresses: boolean;
+
+  /**
+   * The network port ENSIndexer listens for http requests on. ENSIndexer
+   * exposes various APIs, most notably the GRAPHQL API. All APIs are accessible
+   * on the same port. This defaults to DEFAULT_PORT if not specified.
+   *
+   * Invariants:
+   * - The port must be an integer between 1 and 65535
+   */
+  port: number;
+
+  /**
+   * The endpoint URL for the ENSRainbow API. ENSIndexer uses this for fetching
+   * data about labelhashes (healing) when indexing. This should be set to a
+   * colocated instance of ENSRainbow for best performance.
+   *
+   * Invariant: The URL must be a valid URL. localhost urls are allowed,
+   * and expected.
+   */
+  ensRainbowEndpointUrl: string;
+
+  /**
+   * Configuration for each indexed chain, keyed by chain ID.
+   *
+   * Invariants:
+   * - Each key (chainId) must be a number
+   * - For any requested plugin, the config will have a key for chainId in this
+   *   indexedChains object which is ensured by the validateChainConfigs
+   *   function in `validations.ts` and not part of the schema validation here.
+   */
+  indexedChains: Record<number, ChainConfig>;
+
+  /**
+   * The database connection string for the indexer if present. When undefined
+   * ponder will default to using an in-memory database, pglite.
+   *
+   * Invariants:
+   * - The URL must be a valid PostgreSQL connection string or undefined.
+   */
+  databaseUrl: string | undefined;
+}
+
+/**
+ * Represents the raw unvalidated environment variables for a single chain.
+ *
+ * Since an RPC_URL_<chainId> is checked in `getChainsFromEnv`, the value
+ * for rpcEndpointUrl is guaranteed to be a non-empty string.
+ *
+ * rpcMaxRequestsPerSecond is optional and will be undefined if the
+ * RPC_REQUEST_RATE_LIMIT_<chainId> environment variable is not set.
+ */
+export interface RawChainConfig {
```

Comment:
```suggestion
export interface ChainEnvironment {
```

Not sure, just a quick brainstorming idea. I thought it might be nice to align the naming of this interface with the naming of `ENSIndexerEnvironment`.